src/Cli/Microsoft.TemplateEngine.Cli/Commands/SharedOptions.cs

```diff
@@ -26,7 +26,14 @@ public static class SharedOptions
         internal static CliOption<string> NameOption { get; } = new("--name", "-n")
         {
             Description = SymbolStrings.TemplateCommand_Option_Name,
-            Arity = new ArgumentArity(1, 1)
+            Arity = new ArgumentArity(1, 1),
+            Hidden = true
+        };
+
+        internal static CliArgument<string> NameArgument { get; } = new CliArgument<string>("name")
+        {
+            Description = SymbolStrings.TemplateCommand_Option_Name,
+            Arity = new ArgumentArity(0, 1)
```

Comment:
Even though this is a shared name 'argument', it is used in the code the same way as the rest of the shared options.